homeassistant/components/autarco/sensor.py

```diff
@@ -0,0 +1,121 @@
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.sensor import (
+    DOMAIN as SENSOR_DOMAIN,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfEnergy, UnitOfPower
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import AutarcoData, AutarcoDataUpdateCoordinator
+
+
+@dataclass(frozen=True, kw_only=True)
+class AutarcoSensorEntityDescription(SensorEntityDescription):
+    """Describes an Autarco sensor entity."""
+
+    state: Callable[[AutarcoData], Any | None]
+
+
+SENSORS_SOLAR: tuple[AutarcoSensorEntityDescription, ...] = (
+    AutarcoSensorEntityDescription(
+        key="power_production",
+        translation_key="power_production",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        state=lambda data: data.solar.power_production,
+    ),
+    AutarcoSensorEntityDescription(
+        key="energy_production_today",
+        translation_key="energy_production_today",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state=lambda data: data.solar.energy_production_today,
+    ),
+    AutarcoSensorEntityDescription(
+        key="energy_production_month",
+        translation_key="energy_production_month",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state=lambda data: data.solar.energy_production_month,
+    ),
+    AutarcoSensorEntityDescription(
+        key="energy_production_total",
+        translation_key="energy_production_total",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state=lambda data: data.solar.energy_production_total,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Autarco sensors based on a config entry."""
+    coordinator: AutarcoDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    entities: list[AutarcoSensorEntity] = [
+        AutarcoSensorEntity(
+            coordinator=coordinator,
+            description=description,
+            name="Solar",
+            service="solar",
+        )
+        for description in SENSORS_SOLAR
+    ]
+    async_add_entities(entities)
```

Comment:
can be merged and the list can be removed as the `async_add_entities` can get a generator expression